ext/ldap/ldap.c

```diff
@@ -1822,8 +1822,9 @@ PHP_FUNCTION(ldap_modify_batch)
 				oper = LDAP_MOD_REPLACE;
 				break;
 			default:
-				php_error_docref(NULL, E_ERROR, "Unknown and uncaught modification type.");
-				RETURN_FALSE;
+				zend_throw_error(zend_ce_error, "Unknown and uncaught modification type.");
+				RETVAL_FALSE;
+				goto cleanup;
```

Comment:
Fixed by calling `efree(ldap_mods[i])` and setting `num_mods` to `i` before going to `cleanup`.